pyinstrument/profiler.py

```diff
@@ -246,6 +269,19 @@ def __enter__(self):
     def __exit__(self, *args: Any):
         self.stop()
 
+    # pylint: disable=W0613
+    def _sampler_state_changed(
+        self,
+        event: str,
+        thread_id: str,
+            time: float):
+        if not self._active_session:
+            raise RuntimeError(
+                "Received a state change without an active session. Please file an issue on pyinstrument Github describing how you made this happen!"
+            )
+        if event == 'thread_start':
+            self._active_session.record_thread_start(thread_id, time)
+
     # pylint: disable=W0613
     def _sampler_saw_call_stack(
```

Comment:
I'm confused why this method doesn't need to change? Don't we need to separate out the storage of the different threads?

Reply:
The frame records get a root thread id added which allows us to reconstruct what is separate - this is done in build_call_stack

Reply:
(different threads will never clash in the pythons code due to the GIL)